app/kuma-dp/pkg/dataplane/dnsserver/dnsserver.go

```diff
@@ -50,47 +49,12 @@ const DefaultCoreFileTemplate = `.:{{ .CoreDNSPort }} {
     }
 }`
 
-func getSelfPath() (string, error) {
-	ex, err := os.Executable()
-	if err != nil {
-		return "", err
-	}
-
-	return filepath.Dir(ex), nil
-}
-
-func lookupBinaryPath(candidatePaths []string) (string, error) {
-	for _, candidatePath := range candidatePaths {
-		path, err := exec.LookPath(candidatePath)
-		if err == nil {
-			return path, nil
-		}
-	}
-
-	return "", errors.Errorf("could not find binary in any of the following paths: %v", candidatePaths)
-}
-
 func lookupDNSServerPath(configuredPath string) (string, error) {
-	selfPath, err := getSelfPath()
-	if err != nil {
-		return "", err
-	}
-
-	cwd, err := os.Getwd()
-	if err != nil {
-		return "", err
-	}
-
-	path, err := lookupBinaryPath([]string{
-		configuredPath,
-		selfPath + "/coredns",
-		cwd + "/coredns",
-	})
-	if err != nil {
-		return "", err
-	}
-
-	return path, nil
+	return files.LookupBinaryPath(
+		files.LookupInPath(configuredPath),
+		files.LookupInCurrentDirectory("coredns"),
+		files.LookupNextToCurrentExecutable("coredns"),
```

Comment:
Here, and in other places, the previous order was to check the current directory last, which is more reliable and secure.